src/components/Carousel.js

```diff
@@ -107,36 +107,30 @@ class Carousel extends Component {
         this.setupCarousel();
     }
 
-    componentWillReceiveProps(nextProps) {
-        if (nextProps.selectedItem !== this.state.selectedItem) {
-            this.updateSizes();
-            this.moveTo(nextProps.selectedItem);
-        }
-
-        if (nextProps.autoPlay !== this.state.autoPlay) {
-            this.setState(
-                {
-                    autoPlay: nextProps.autoPlay,
-                },
-                () => {
-                    if (this.state.autoPlay) {
-                        this.setupAutoPlay();
-                    } else {
-                        this.destroyAutoPlay();
-                    }
-                }
-            );
-        }
-    }
-
     componentDidUpdate(prevProps, prevState) {
         if (!prevProps.children && this.props.children && !this.state.initialized) {
             this.setupCarousel();
         }
+
         if (prevState.swiping && !this.state.swiping) {
             // We stopped swiping, ensure we are heading to the new/current slide and not stuck
             this.resetPosition();
         }
+
+        if (prevProps.selectedItem !== this.props.selectedItem) {
```

Comment:
believe the logic here should actually be (since the component has updated at this point), if `this.props.selectedItem` does not match `this.state.selectedItem`, but this is a very confusing thing to write…